### PR TITLE
Add 2.7 support

### PR DESCRIPTION
**Pull request by mbj**
* But also fix all bugs discovered from using parsers tests to discover
  the 2.7 support.

### CODE REVIEW

**Comment by mbj** [comment]
This is the most bananas Ruby code I recently wrote.

It (locally) passes its own tests, and the corpus. Currently still relies on accessing `ruby27.y` from the parser repo to load the operator precedences.

Also the code is not yet reduced. Basically I smashed any conditional in there to get the tests to pass. I've to do a (mutant assisted) pass to reduce the code .

Overall unparser is now *much* better, I removed the cheats ;)

**Comment by mbj** [comment]
> It (locally) passes its own tests, and the corpus. Currently still relies on accessing `ruby27.y` from the parser repo to load the operator precedences.

This is fixed, generates a data file in `lib/unparser/precedence/data.rb` now from `scripts/dump-precedence.rb`, the `rubyX.y` are still source of truth.

**Comment by mbj** [comment]
> This is fixed, generates a data file in `lib/unparser/precedence/data.rb` now from `scripts/dump-precedence.rb`, the `rubyX.y` are still source of truth.

Lol turns out that I do not even need the precedence data, mutant successively taught me that I can get away with a far simpler signal.

File: lib/unparser/emitter/binary.rb [hunk 1]
```diff
@@ -2,33 +2,20 @@
 
 module Unparser
   class Emitter
-    # Base class for binary emitters
+    # Non send binary operator / keywrod emitter
```
Comment (dgollahon) [inline lib/unparser/emitter/binary.rb:5]: typo: `keywrod`

File: lib/unparser/emitter/rescue.rb [hunk 2]
```diff
@@ -2,111 +2,15 @@
 
 module Unparser
   class Emitter
-    # Emitter for rescue nodes
+    # Emitter for alias nodes
```
Comment (dgollahon) [inline lib/unparser/emitter/rescue.rb:5]: I think this should still say `rescue`?

File: lib/unparser/emitter/xstr.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module Unparser
+  class Emitter
+    # Dynamic symbol literal emitter
```
Comment (dgollahon) [inline lib/unparser/emitter/xstr.rb:5]: Seems to be a copy/paste from elsewhere.

**Comment by mbj** [comment]
@dgollahon Thanks for all your comments, all fixed.

**Comment by mbj** [comment]
This now passes mutant on `Unparser::{Emmiter,Writer}*` expressions. Minus exclusions that are mostly from mutant crashing MRI and mutant trying to unparse arg nodes as root. Which is not supported anymore by this unparser.

Mutant will update to the lastest unparser and than the exclusions can be removed.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -46,34 +46,14 @@ jobs:
...
-    - uses: actions/checkout@v2
-      with:
-        fetch-depth: 0
-    - uses: actions/setup-ruby@v1
-      with:
-        ruby-version: ${{ matrix.ruby }}
-    - run: |
-        gem install bundler
-        bundle install
-    - run: ./mutant.sh --since HEAD~1 -- 'Unparser*'
```
Comment (mbj) [inline .github/workflows/ci.yml:68]: I've yet to re-introduce mutant on CI.

**Pull request merged**